l3kernel/l3regex.dtx

```diff
@@ -1980,7 +1980,7 @@
 %   but printable ascii, from space (hex \texttt{20}) to
 %   \texttt{del} (hex \texttt{7E}).
 %    \begin{macrocode}
-\prg_new_conditional:Npnn \@@_char_if_special:N #1 { TF }
+\prg_new_conditional:Npn \@@_char_if_special:N #1
```

Comment:
This is really the core question (and why this is a PR not a direct change) :) If we want to keep the `Npnn` version for the kernel, the same argument applies to any programmer. So we would then want to adjust `\prg_generate_conditional_variant:Nnn` to check that we are creating variants of known branching forms (cf. the original issue report). On the other hand, if we go with 'we can afford the tokens' then this PR makes sense - but it really needs to apply all-round. That's my take, at least.

Reply:
well, if a programmer has a private conditional he or she has full control about which variants are needed, so it seems ok to me to offer just generating those that are required (since nobody else should use the conditional or generate further variants. For public ones, using the command that generates all should be recommended but for private once one could leave it up to the programmer what to do. I don't think it requies checking if what happens is allowed, because it is solely under the programmers control, but that's just my feeling.

Reply:
I see what you are getting at, but that still requires that one tracks what is happening with 'internal' conditionals. I'm wondering if for that it would be better simply to fix `\prg_generate_conditional_variant:Nnn` to do the check that is required ...

Reply:
> I see what you are getting at, but that still requires that one tracks what is happening with 'internal' conditionals. 

why is tracking what happens with internal conditionals required?

Reply:
I meant that the programmer has to know what variants they've defined

Reply:
ah yes, just like he or she needs to remember what the commands are called that she defined, not really different or  is it?